lib/app.js

```diff
@@ -0,0 +1,39 @@
+
+const ChatRoom = require('./chatRoom');
+const parseMessage = require('./parseMessage');
+const { createServer } = require('net');
+
+module.exports = () => {
+  const chatRoom = new ChatRoom();
+ 
+  return createServer(client => {
+  /* eslint-disable-next-line */
+  console.log('client connected');
+    client.setEncoding('utf8');
+    chatRoom.add(client);
+
+    client.on('end', () => chatRoom.remove(client));
+  
+    client.on('data', data => {
+      const message = parseMessage(data);
+      if(!message) return; {
+        switch(message.command) {
+          case 'all':
+            chatRoom
+              .broadcast(client)
+              .forEach(c => c.write(`${client.username}: ${message.text}`));
+            break;
+          case 'nick':
+            chatRoom.rename(client.username, message.arg);
+            break;
+          case 'dm':
+            chatRoom.getClient(message.arg).write(`(dm) ${client.username}: ${message.text}`);
+          // eslint-disable-next-line no-fallthrough
```

Comment:
```suggestion
            break;
```
When you "DM" someone, the user sending the DM gets "Unknown command" returned even though the message goes through. Adding a quick 'break' to the switch case ensures it doesn't fall through to the default after executing DM.